examples/example_evaluate_in_c.py

```diff
@@ -0,0 +1,107 @@
+"""
+Example for evolutionary regression, with evaluation in c
+=========================================================
+"""
+
+# The docopt str is added explicitly to ensure compatibility with
+# sphinx-gallery.
+docopt_str = """
+   Usage:
+     example_evaluate_in_c.py
+
+   Options:
+     -h --help
+"""
+
+import pathlib
+import subprocess
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+# %%
+# We first define a helper function for compiling the c code. It creates
+# object files from the file and main script and creates an executable
+
+
+def compile_c_code(filename, scriptname, path):
+
+    # assert all necessary files exist
+    path_file_c = pathlib.Path(f"{path}/{filename}.c")
+    path_file_h = pathlib.Path(f"{path}/{filename}.h")
+    path_script_c = pathlib.Path(f"{path}/{scriptname}.c")
+    path_script_h = pathlib.Path(f"{path}/{scriptname}.h")
+    assert (
+        path_file_c.is_file()
+        & path_file_h.is_file()
+        & path_script_c.is_file()
+        & path_script_h.is_file()
+    )
+
+    # compile file with rule
+    subprocess.run(["gcc", "-c", "-fPIC", f"{path}/{filename}.c", "-o", f"{path}/{filename}.o"])
```

Comment:
```suggestion
    subprocess.check_call(["gcc", "-c", "-fPIC", f"{path}/{filename}.c", "-o", f"{path}/{filename}.o"])
```
this makes noise when the compilation fails. same for the other calls